### PR TITLE
Use secure cookie if https

### PR DESCRIPTION
**Pull request by ludeeus**
## Breaking change

Follow up to https://github.com/home-assistant/frontend/pull/6556#discussion_r471483696

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: test-mocha/hassio/create_session.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    global.document = {};
+    // @ts-ignore
+    global.location = { protocol: "https:" };
+    await createHassioSession({
+      // @ts-ignore
+      callApi: async function () {
+        return { data: { session: sessionID } };
+      },
+    });
+    assert.deepEqual(
```
Comment (bramkragten) [inline test-mocha/hassio/create_session.spec.ts:35]: No need for deepEqual as it is just a string compare?

File: test-mocha/hassio/create_session.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    global.document = {};
+    // @ts-ignore
+    global.location = {};
+    await createHassioSession({
+      // @ts-ignore
+      callApi: async function () {
+        return { data: { session: sessionID } };
+      },
+    });
+    assert.deepEqual(
```
Comment (bramkragten) [inline test-mocha/hassio/create_session.spec.ts:18]: No need for deepEqual as it is just a string compare?

File: test-mocha/hassio/create_session.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as assert from "assert";
+import { createHassioSession } from "../../src/data/hassio/supervisor";
+
+const sessionID = "fhdsu73rh3io4h8f3irhjel8ousafehf8f3yh";
+
+describe("Create hassio session", function () {
```
Comment (bramkragten) [inline test-mocha/hassio/create_session.spec.ts:6]: Tests? 😱 :tada:

File: test-mocha/hassio/create_session.spec.ts (hunk 1 above)
Comment (ludeeus) [reply test-mocha/hassio/create_session.spec.ts:35]: True

File: test-mocha/hassio/create_session.spec.ts (hunk 2 above)
Comment (ludeeus) [reply test-mocha/hassio/create_session.spec.ts:18]: True

File: test-mocha/hassio/create_session.spec.ts (hunk 3 above)
Comment (ludeeus) [reply test-mocha/hassio/create_session.spec.ts:6]: Yeah :man_shrugging: trying something new I guess :joy:

**Review by bramkragten — APPROVED** [review]

**Pull request merged**